mlflow/utils/proto_json_utils.py

```diff
@@ -187,6 +189,8 @@ def encode_binary(x):
             return np.datetime_as_string(o), True
         if isinstance(o, (pd.Timestamp, datetime.date, datetime.datetime, datetime.time)):
             return o.isoformat(), True
+        if isinstance(o, pydantic.BaseModel):
+            return o.model_dump() if IS_PYDANTIC_V2_OR_NEWER else o.dict(), True
```

Comment:
note (unrelated to this PR): we should probably define a common function `pydantic.BaseModel -> dict` in order to refactor the code easily when we stop supporting pedantic v1.

Reply:
Agree, we need to put the function in a common utils file that's accessible everywhere